controllers/v1/reportsController.js

```diff
@@ -1549,7 +1549,27 @@ module.exports = class Reports {
         };
 
         const submissionIds = await database.models.submissions.find(queryParams, {
-          _id: 1
+          _id: 1, evaluationFrameworkId: 1
+        })
+
+        let evaluationIds = [];
+        submissionIds.forEach(eachSubmission => {
+          evaluationIds.push(eachSubmission._doc.evaluationFrameworkId)
+        })
+
+        const programsDocument = await database.models.programs.find({
+          "components": {
+            $elemMatch: {
+              "id": { $in: evaluationIds }
+            }
+          }
+        }, { "components.schoolProfileFieldsPerSchoolTypes": 1 })
+
+        let schoolProfileFields;
+        programsDocument.forEach(eachProgram => {
+          eachProgram.components.forEach(eachComponent => {
+            schoolProfileFields = this.getSchoolProfileFields(eachComponent.schoolProfileFieldsPerSchoolTypes)
```

Comment:
No need foreach. Also add "await" so that it goes forward only once it gets school fields. @aman-tunerlabs